dotnet-desktop-guide/net/wpf/properties/xaml-loading-and-dependency-properties.md

```diff
@@ -0,0 +1,56 @@
+
+## Prerequisites
+
+The article assumes a basic knowledge of dependency properties, and that you've read [Dependency properties overview](dependency-properties-overview.md). To follow the examples in this article, it helps if you're familiar with Extensible Application Markup Language (XAML) and know how to write WPF applications.
+
+## WPF XAML loader performance
+
+It's computationally less expensive for the WPF XAML processor to directly call <xref:System.Windows.DependencyObject.SetValue%2A> to set the value of a dependency property, rather than use the property wrapper of a dependency property.
+
+If the XAML processor did use the property wrapper, it would require inferring the entire object model of the backing code based only on the type and member relationships indicated in markup. Although the type can be identified from markup by using a combination of `xmlns` and assembly attributes, identifying the members, determining which members can be set as an attribute, and resolving supported property value types, would require extensive reflection using <xref:System.Reflection.PropertyInfo>.
+
+The WPF property system maintains a storage table of dependency properties implemented on a given <xref:System.Windows.DependencyObject> derived type. The XAML processor uses that table to infer the dependency property identifier for a dependency property. For example, by convention the dependency property identifier for a dependency property named `ABC` is `ABCProperty`. The XAML processor can efficiently set the value of any dependency property by calling the `SetValue` method on its containing type using the dependency property identifier.
+
+For more information on dependency property wrappers, see [Custom dependency properties](custom-dependency-properties.md).
+
+## Implications for custom dependency properties
+
+The WPF XAML processor bypasses property wrappers and directly calls <xref:System.Windows.DependencyObject.SetValue%2A> to set a dependency property value. So, avoid putting any extra logic in the `set` accessor of your custom dependency property because that logic won't run when a property value is set in XAML. The `set` accessor should only contain a `SetValue` call.
+
+Similarly, aspects of the WPF XAML processor that get property values bypass the property wrapper and directly call <xref:System.Windows.DependencyObject.GetValue%2A>. So, also avoid putting any extra logic in the `get` accessor of your custom dependency property because that logic won't run when a property value is read in XAML. The `get` accessor should only contain a `GetValue` call.
+
+## Dependency property with wrapper example
+
+The following example shows a recommended dependency property definition with property wrappers. The dependency property identifier is stored as a `public static readonly` field, and the `get` and `set` accessors contain no code beyond the necessary WPF property system methods that back the dependency property value.
```

Comment:
This is great. I think it would be very beneficial to mention the that because you shouldn't modify the `set` accessor beyond what is necessary for the WPF property system, the alternative is to use the `PropertyChangedCallback` to run code when the value is set. A cross link to https://review.docs.microsoft.com/en-us/dotnet/desktop/wpf/properties/dependency-property-callbacks-and-validation?view=netdesktop-6.0&branch=pr-en-us-1257#property-changed-callbacks would be appropriate.

Reply:
Great idea, thank you. I've expanded the snippet to include the property changed callback as well.